packages/web/src/pages/oauth-login-page/utils.ts

```diff
@@ -251,3 +272,107 @@ export const validateWriteOnceParams = ({
   }
   return { error, txParams }
 }
+
+export const handleAuthorizeConnectDashboardWallet = async ({
+  state,
+  originUrl,
+  onError,
+  account,
+  txParams
+}: {
+  state: string | string[] | null
+  originUrl: URL | null
+  onError: ({
+    isUserError,
+    errorMessage,
+    error
+  }: {
+    isUserError: boolean
+    errorMessage: string
+    error?: Error
+  }) => void
+  account: User
+  txParams: ConnectDashboardWalletParams
+}) => {
+  if (!window.opener || !originUrl) {
+    onError({
+      isUserError: false,
+      errorMessage: messages.noWindowError
+    })
+    return false
+  }
+
+  let resolveWalletSignature:
+    | ((value: { message: string; signature: string }) => void)
+    | null = null
+  const receiveWalletSignaturePromise = new Promise<{
+    message: string
+    signature: string
+  }>((resolve) => {
+    resolveWalletSignature = resolve
+  })
+  window.addEventListener(
```

Comment:
Does this/should this be cleaned up at some point?

Reply:
good call, ill clean it up!